File: php/src/Behavioral/State/DvdStateNameStars.php
<?

#//DvdStateNameStars - two of two Concrete States

require_once 'DvdStateName.php';
require_once 'DvdStateNameExclaim.php';

class DvdStateNameStars implements DvdStateName
{

	public $starCount;

	function __construct()
	{
		$this->starCount = 0;
	}
		 
	function getName($ctx, $name)
	{
		$t = $name;
		$t = str_replace(' ','*', $t);
		# show stars twice, switch to exclamation point
		$this->starCount++;
		if($this->starCount > 1)
			$ctx->setDvdStateName(new DvdStateNameExclaim());
		return $t;
	}
}

File: python/src/Behavioral/State/DvdStateNameStars.py
# DvdStateNameStars- two of two Concrete States


from DvdStateName import DvdStateName

class DvdStateNameStars(DvdStateName):

	starCount = 0

	def __init__(self,starCount = 0):
		self.starCount = starCount
	
	def showName (self,ctx, name):
		self.starCount += 1
		retval = name.replace(' ','*', self.starCount)  #// show stars twice, switch to exclamation point
		if self.starCount > 1 :
			print "made switch to exclaim"
			ctx.setDvdStateName( DvdStateNameExclaim())
		return retval;

from DvdStateNameExclaim import DvdStateNameExclaim
